jOOQ/src/main/java/org/jooq/impl/InsertQueryImpl.java

```diff
@@ -248,30 +273,41 @@ final void accept0(Context<?> ctx) {
                 case POSTGRES: {
                     toSQLInsert(ctx);
                     ctx.formatSeparator()
-                       .start(INSERT_ON_DUPLICATE_KEY_UPDATE)
-                       .visit(K_ON_CONFLICT)
-                       .sql(" (");
+                       .start(INSERT_ON_DUPLICATE_KEY_UPDATE);
 
-                    if (onConflict != null && onConflict.size() > 0) {
-                        boolean qualify = ctx.qualify();
+                    if (onConstraint != null) {
+                        ctx.data(DATA_CONSTRAINT_REFERENCE, true);
+                        ctx.visit(K_ON_CONSTRAINT)
```

Comment:
I suspect the `ON CONSTRAINT` keywords here should be prefixed by `ON CONFLICT`. The current implementation seems to produce a syntax error. I'll fix this when implementing the relevant integration tests.